src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/19.diff.sql

```diff
@@ -0,0 +1,302 @@
+                            ON pf.function_id = prv.function_id
+                    WHERE pf.name = N'PartitionFunction_ResourceChangeData_Timestamp'
+                        and CONVERT(datetime2(7), prv.value, 126) = @rightPartitionBoundary) 
+        BEGIN
+            -- Creates new empty partition by creating new boundary value and specifying NEXT USED file group.
+            ALTER PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp NEXT USED [Primary];
+            ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() SPLIT RANGE(@rightPartitionBoundary);
+        END;
+            
+        SET @numberOfPartitions -= 1;
+    END;
+END;
+
+-- Creates a staging table 
+IF NOT EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeDataStaging')
+BEGIN
+    -- Staging table that will be used for partition switch out. 
+    CREATE TABLE dbo.ResourceChangeDataStaging 
+    (
+        Id bigint IDENTITY(1,1) NOT NULL,
+        Timestamp datetime2(7) NOT NULL CONSTRAINT DF_ResourceChangeDataStaging_Timestamp DEFAULT sysutcdatetime(),
+        ResourceId varchar(64) NOT NULL,
+        ResourceTypeId smallint NOT NULL,
+        ResourceVersion int NOT NULL,
+        ResourceChangeTypeId tinyint NOT NULL,
+        CONSTRAINT PK_ResourceChangeDataStaging_TimestampId PRIMARY KEY (Timestamp, Id)
+    ) ON [PRIMARY]
+END;
+GO
+
+-- Adds a check constraint on the staging table for a partition boundary validation.
+IF NOT EXISTS(SELECT 1 FROM sys.check_constraints WHERE name = 'CHK_ResourceChangeDataStaging_partition')
+BEGIN    
+    ALTER TABLE dbo.ResourceChangeDataStaging WITH CHECK 
+        ADD CONSTRAINT CHK_ResourceChangeDataStaging_partition CHECK(Timestamp < CONVERT(DATETIME2(7), '9999-12-31 23:59:59.9999999'));
+END;
+GO
+
+IF EXISTS(SELECT 1 FROM sys.check_constraints WHERE name = 'CHK_ResourceChangeDataStaging_partition')
+BEGIN    
+    ALTER TABLE dbo.ResourceChangeDataStaging CHECK CONSTRAINT CHK_ResourceChangeDataStaging_partition;
+END;
+GO
+
+    
+IF EXISTS (SELECT * FROM sys.indexes WHERE name = 'PK_ResourceChangeData')
+BEGIN
+    -- Drops index.
+    ALTER TABLE dbo.ResourceChangeData DROP CONSTRAINT PK_ResourceChangeData WITH (ONLINE = OFF);
+END;
+        
+IF NOT EXISTS (SELECT * FROM sys.indexes WHERE name = 'PK_ResourceChangeData_TimestampId')
+BEGIN    
+    -- Adds primary key clustered index.
+    ALTER TABLE dbo.ResourceChangeData ADD CONSTRAINT PK_ResourceChangeData_TimestampId
+        PRIMARY KEY CLUSTERED(Timestamp ASC, Id ASC) ON PartitionScheme_ResourceChangeData_Timestamp(Timestamp)
+END;
+GO
+
+/*************************************************************
+    Purge partition feature for resource change data
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     ConfigurePartitionOnResourceChanges
+--
+-- DESCRIPTION
+--     Creates initial partitions for future datetimes on the resource change data table.
```

Comment:
As discussed, specify that this will only add `@numberOfFuturePartitionsToAdd` future partitions if they do not already exist.

Reply:
Added a comment.